### ISSUE TITLE
Map keys are not evaluated

### ISSUE THREAD

**Issue by OnkelTem** [sass/libsass#1086]
This code produces error
```sass
$map: (-1px: 12);
```
Error message: 
```
  undefined:undefined  _Map_base::at
```
Also note 'undefined:undefined' standing for the line and column number — they are missed. 

Using node-sass v3.0.0-beta.5 (pointing to libsass v3.2.0-beta.5)

I believe this temporary workaround can be used:
```sass
$minus1px: -1px;
$map: ($minus1px: 12);
```

**Comment by mgreter** [comment]
This is actually not really a regression, although that exact case was working before. The reason for this is, that things like the following did not work in libsass 3.1.0:

```scss
$map: (0 - 1px: 12);
foo { bar: inspect($map); }
```

We changed the way we parse negative numbers to wrap them into an unary_operator. This is more in line with the actual meaning and solved problems in other places, but unfortunately brings up another problem to the surface here. IMO we need to eval the keys before "using" them.

**Comment by mgreter** [comment]
This specific issue (and a few more) should be fixed by #1090. I'm quite sure that we have more subtile problems with map handlings, but this should at least fix quite a lot of the more common use-cases!
